server/aesdsocket.c

```diff
@@ -202,21 +208,48 @@ void *new_thread_fn( void *t_parameters )
         count++;
     }
 
-    lseek(aesdsocketdata_fd, 0, SEEK_END);
+//    lseek(aesdsocketdata_fd, 0, SEEK_END);
 
-    status = pthread_mutex_lock(&mutex);
+#ifndef USE_AESD_CHAR_DEVICE
+    int status = pthread_mutex_lock(&mutex);
     if( status != 0 )
     {
         perror("mutex lock");
     }
-    status= write(aesdsocketdata_fd, recieved_packets, recv_buff_len );
-    if( -1 == status)
+#endif
+	//open the socket data file
+	int aesdsocketdata_fd= open(AESDSOCKET_FILEPATH,O_RDWR | O_CREAT | O_APPEND,0666);
+
+	if( -1 == aesdsocketdata_fd ){
+	
+		syslog(LOG_ERR, "not able to open file %s",AESDSOCKET_FILEPATH);
+		
+	}
+	
+    if(0 == strncmp(recieved_packets, ioctl_string, strlen(ioctl_string)))
+    {
+        struct aesd_seekto seek_info; 
+        
+        sscanf(recieved_packets, "AESDCHAR_IOCSEEKTO:%d,%d", &seek_info.write_cmd, &seek_info.write_cmd_offset);
```

Comment:
@akshithaluguri22 what happens if sscanf fails here?  What will be passed to ioctl?